front/src/modules/simulationResult/components/ManchetteWithSpaceTimeChart/utils.ts

```diff
@@ -20,22 +22,39 @@ const getPathStyle = (
     backgroundColor?: string;
   };
 } => {
-  if (hovered && 'pathId' in hovered.element && path.id === hovered?.element.pathId && !dragging) {
-    return { color: PATH_COLORS.HOVERED_PATH, level: 1 };
+  const trainById = Object.fromEntries(
+    timetableItemsWithDetails?.map((t) => [t.id, { category: t.category }]) ?? []
+  );
```

Comment:
It seems like we're building a map just to access a single element. It would be simpler to just use `Array.find()` here.